sql/src/main/java/io/crate/execution/engine/window/WindowProjector.java

```diff
@@ -98,19 +99,20 @@ public static WindowProjector fromProjection(WindowAggProjection projection,
             FunctionImplementation impl = functions.getQualified(
                 windowFunctionContext.function().info().ident());
             if (impl instanceof AggregationFunction) {
-                Input<Boolean> filter;
+                ExpressionsInput<Row, Boolean> filter;
 
                 if (windowFunctionContext.filter().symbolType() == SymbolType.LITERAL) {
-                    //noinspection unchecked
-                    filter = (Input<Boolean>) windowFunctionContext.filter();
+                    filter = new ExpressionsInput<>(Literal.BOOLEAN_TRUE, List.of());
                 } else {
-                    ctx.add(windowFunctionContext.filter());
                     //noinspection unchecked
-                    filter = (Input<Boolean>) ctx.expressions().get(ctx.expressions().size() - 1);
+                    Input<Boolean> filterInput =
+                        (Input<Boolean>) ctx.add(windowFunctionContext.filter());
```

Comment:
Do you think it is worth creating a separate `ctx` instance for the `filter` so that the `expressions` doesn't contain any expressions that are related to the `windowFunctionContext.inputs()` that are added earlier?